common/icap_test.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2015 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package common
+
+import "testing"
+
+/* More test vectors:
+https://github.com/ethereum/web3.js/blob/master/test/iban.fromAddress.js
+https://github.com/ethereum/web3.js/blob/master/test/iban.toAddress.js
+https://github.com/ethereum/web3.js/blob/master/test/iban.isValid.js
+https://github.com/ethereum/libethereum/blob/develop/test/libethcore/icap.cpp
+*/
+
+func TestICAP1(t *testing.T) {
+	addr := HexToAddress("0x52dc504a422f0e2a9e7632a34a50f1a82f8224c7")
+	icap := "XE499OG1EH8ZZI0KXC6N83EKGT1BM97P2O7"
+	decodeEncodeTest(addr, icap, t)
+}
+
+func TestICAP2(t *testing.T) {
+	addr := HexToAddress("0x11c5496aee77c1ba1f0854206a26dda82a81d6d8")
+	icap := "XE1222Q908LN1QBBU6XUQSO1OHWJIOS46OO"
+	decodeEncodeTest(addr, icap, t)
+}
+
+func TestICAPDoubleZeroPrefix(t *testing.T) {
+	addr := HexToAddress("0x0000a5327eab78357cbf2ae8f3d49fd9d90c7d22")
+	icap := "XE0600DQK33XDTYUCRI0KYM5ELAKXDWWF6"
+	decodeEncodeTest(addr, icap, t)
+}
+
+func TestICAPInvalidCheckSum(t *testing.T) {
+	failedDecodingTest("XE7438O073KYGTWWZN0F2WZ0R8PX5ZPPZS", t)
+}
+
+func TestICAPInvalidCountryCode(t *testing.T) {
+	failedDecodingTest("XD7338O073KYGTWWZN0F2WZ0R8PX5ZPPZS", t)
+}
+
+func TestInvalidLength36(t *testing.T) {
+	failedDecodingTest("XE499OG1EH8ZZI0KXC6N83EKGT1BM97P2O77", t)
+}
+func TestInvalidLength33(t *testing.T) {
+	failedDecodingTest("XE499OG1EH8ZZI0KXC6N83EKGT1BM97P2", t)
+}
```

Comment:
These tests look like they could be made [table-driven](https://github.com/golang/go/wiki/TableDrivenTests) with no extra effort.